### PR TITLE
Backport to 6.3 - Navigation: Load the raw property on the navigation fallback 

### PR DESCRIPTION
**Pull request by ramonjd**
Backports the following Gutenberg PR:

- https://github.com/WordPress/gutenberg/pull/52758

cc @scruffian @getdave 

Trac ticket: https://core.trac.wordpress.org/ticket/58557

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php [hunk 1]
```diff
@@ -143,6 +143,49 @@ public function test_adds_links() {
...
+		$links    = $response->get_links();
+
+		// Extract the navigation endpoint URL from the response.
+		$embedded_navigation_href = $links['self'][0]['href'];
+		preg_match( '/\?rest_route=(.*)/', $embedded_navigation_href, $matches );
+		$navigation_endpoint = $matches[1];
+
+		// Fetch the "linked" navigation post from the endpoint, with the context parameter set to 'embed' to simulate fetching embedded links.
+		$request = new WP_REST_Request( 'GET', $navigation_endpoint );
+		$request->set_param( 'context', 'embed' );
```
Comment (ramonjd) [inline tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php:173]: I've ported this test as-is. 

It failed locally unless I changed this line to:

```
$request->set_param( 'context', 'edit' );
```

@getdave @scruffian Any ideas?

File: src/wp-includes/navigation-fallback.php [hunk 2]
```diff
@@ -22,20 +22,35 @@
...
+function wp_add_fields_to_navigation_fallback_embedded_links( $schema ) {
 	// Expose top level fields.
 	$schema['properties']['status']['context']  = array_merge( $schema['properties']['status']['context'], array( 'embed' ) );
 	$schema['properties']['content']['context'] = array_merge( $schema['properties']['content']['context'], array( 'embed' ) );
 
-	// Expose sub properties of content field.
+	/*
+	 * Exposes sub properties of content field.
+	 * These sub properties aren't exposed by the posts controller by default,
+	 * only when the post type attribute is `'editor'`.
```
Comment (ramonjd) [inline src/wp-includes/navigation-fallback.php]: Not sure if what I wrote here is right.

File: src/wp-includes/navigation-fallback.php [hunk 3]
```diff
@@ -22,20 +22,35 @@
...
+	 * @see WP_REST_Posts_Controller::get_item_schema()
+	 */
 	$schema['properties']['content']['properties']['raw']['context']           = array_merge( $schema['properties']['content']['properties']['raw']['context'], array( 'embed' ) );
 	$schema['properties']['content']['properties']['rendered']['context']      = array_merge( $schema['properties']['content']['properties']['rendered']['context'], array( 'embed' ) );
 	$schema['properties']['content']['properties']['block_version']['context'] = array_merge( $schema['properties']['content']['properties']['block_version']['context'], array( 'embed' ) );
 
+	/*
+	 * Exposes sub properties of title field.
+	 * These sub properties aren't exposed by the posts controller by default,
+	 * only when the post type attribute is `'editor'`.
```
Comment (ramonjd) [inline src/wp-includes/navigation-fallback.php]: Here too

File: tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php (hunk 1 above)
Comment (ramonjd) [reply tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php:173]:
```
1) WP_REST_Navigation_Fallback_Controller_Test::test_embedded_navigation_post_contains_required_fields
Response title should contain a "status" field.
Failed asserting that an array has the key 'status'.
```

File: tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php [hunk 4]
```diff
@@ -143,6 +143,49 @@ public function test_adds_links() {
...
+	/**
+	 * Tests that the correct filters are applied to the context parameter.
+	 *
+	 * By default, the REST response for the Posts Controller will not return all fields
+	 * when the context is set to 'embed'. Assert that correct additional fields are added
+	 * to the embedded Navigation Post, when the navigation fallback endpoint
+	 * is called with the `_embed` param.
+	 *
+	 * @ticket 58557
+	 * @covers wp_add_fields_to_navigation_fallback_embedded_links
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php]:
```suggestion
	 * @ticket 58557
	 *
	 * @covers wp_add_fields_to_navigation_fallback_embedded_links
```

File: tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php (hunk 1 above)
Comment (getdave) [reply tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php:173]: If it doesn't work it probably means that the filter on `rest_wp_navigation_item_schema` isn't being called. Is the file being included?

The whole point of the filter is that if you make a request for a single Navigation Menu and change the context to `embed` you will get these additional fields.

That's what the test does.

File: src/wp-includes/navigation-fallback.php (hunk 2 above)
Comment (getdave) [inline src/wp-includes/navigation-fallback.php]:
```suggestion
	 * These sub properties aren't exposed by the posts controller by default,
	 * for requests where context is `embed`.
```

File: src/wp-includes/navigation-fallback.php (hunk 3 above)
Comment (getdave) [inline src/wp-includes/navigation-fallback.php]:
```suggestion
	 * for requests where context is `embed`.
```

File: tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php (hunk 1 above)
Comment (ramonjd) [reply tests/phpunit/tests/rest-api/rest-navigation-fallback-controller.php:173]: Thanks @getdave 

You know what: it wasn't imported 🤣 

The file had already been backported, so I assumed it was.

But it wasn't.

So here we are. 

Thanks for the tip!

File: src/wp-settings.php [hunk 5]
```diff
@@ -347,6 +347,7 @@
 require ABSPATH . WPINC . '/block-supports/spacing.php';
 require ABSPATH . WPINC . '/block-supports/typography.php';
 require ABSPATH . WPINC . '/block-supports/settings.php';
+require ABSPATH . WPINC . '/navigation-fallback.php';
```
Comment (getdave) [inline src/wp-settings.php:350]: Sorry about this! Thank you for fixing.

Comment (ramonjd) [reply src/wp-settings.php:350]: All's well that ends well. 🍺

**Review by scruffian — APPROVED** [review]
This is looking good now.

File: src/wp-settings.php (hunk 5 above)
Comment (getdave) [reply src/wp-settings.php:350]: This is an(other) lesson in why tests are _essential_.

Comment (ramonjd) [reply src/wp-settings.php:350]: You said it!

**Review by tellthemachines — APPROVED** [review]
Changes LGTM!

**Comment by tellthemachines** [comment]
Committed in [r56296](https://core.trac.wordpress.org/changeset/56296) to trunk / [r56297](https://core.trac.wordpress.org/changeset/56297) to 6.3.

**Pull request closed without merging**